diff --git a/kernel/irq/proc.c b/kernel/irq/proc.c
index 29c2404e7..7e5a1aad8 100644
--- a/kernel/irq/proc.c
+++ b/kernel/irq/proc.c
@@ -447,6 +447,21 @@ int __weak arch_show_interrupts(struct seq_file *p, int prec)
 # define ACTUAL_NR_IRQS irq_get_nr_irqs()
 #endif
 
+static void get_interrupt_info(struct irqaction *action_p)
+{
+	unsigned int irq_num = action_p->irq;
+	void * irq_handler = NULL;
+
+	if (action_p->handler) {
+		irq_handler = (void*) action_p->handler;
+	}
+
+	if (irq_handler) {
+		trace_printk("[%s] %d: %s, irq_handler: %pS \n",
+			     current->comm, irq_num, action_p->name, irq_handler);
+	}
+}
+
 int show_interrupts(struct seq_file *p, void *v)
 {
 	const unsigned int nr_irqs = irq_get_nr_irqs();
@@ -511,6 +526,11 @@ int show_interrupts(struct seq_file *p, void *v)
 		seq_printf(p, "-%-8s", desc->name);
 
 	action = desc->action;
+
+	if (action) {
+		get_interrupt_info(action);
+	}
+
 	if (action) {
 		seq_printf(p, "  %s", action->name);
 		while ((action = action->next) != NULL)
